src/main/java/appeng/fluids/parts/PartFluidStorageBus.java

```diff
@@ -0,0 +1,508 @@
+		}
+		catch( final GridAccessException e )
+		{
+			// :P
+		}
+	}
+
+	@Override
+	public boolean onPartActivate( final EntityPlayer player, final EnumHand hand, final Vec3d pos )
+	{
+		if( !player.isSneaking() )
+		{
+			if( Platform.isClient() )
+			{
+				return true;
+			}
+			Platform.openGUI( player, this.getHost().getTile(), this.getSide(), GuiBridge.GUI_STORAGEBUS_FLUID );
+			return true;
+		}
+
+		return false;
+	}
+
+	@Override
+	public void onChangeInventory( final IItemHandler inv, final int slot, final InvOperation mc, final ItemStack removedStack, final ItemStack newStack )
+	{
+		super.onChangeInventory( inv, slot, mc, removedStack, newStack );
+
+		if( inv == this.config )
+		{
+			this.resetCache( true );
+		}
+	}
+
+	@Override
+	public IItemHandler getInventoryByName( final String name )
+	{
+		if( name.equals( "config" ) )
+		{
+			return this.config;
+		}
+
+		return super.getInventoryByName( name );
+	}
+
+	@Override
+	public void readFromNBT( final NBTTagCompound data )
+	{
+		super.readFromNBT( data );
+		this.config.readFromNBT( data, "config" );
+	}
+
+	@Override
+	public void writeToNBT( final NBTTagCompound data )
+	{
+		super.writeToNBT( data );
+		this.config.writeToNBT( data, "config" );
+	}
+
+	@Override
+	public boolean isValid( final Object verificationToken )
+	{
+		return this.handler == verificationToken;
+	}
+
+	@Override
+	public void postChange( final IBaseMonitor<IAEFluidStack> monitor, final Iterable<IAEFluidStack> change, final IActionSource source )
+	{
+		try
+		{
+			if( this.getProxy().isActive() )
+			{
+				this.getProxy().getStorage().postAlterationOfStoredItems( AEApi.instance().storage().getStorageChannel( IFluidStorageChannel.class ), change, this.source );
+			}
+		}
+		catch( final GridAccessException e )
+		{
+			// :(
+		}
+	}
+
+	public MEInventoryHandler<IAEFluidStack> getInternalHandler()
+	{
+		if( this.cached )
+		{
+			return this.handler;
+		}
+
+		final boolean wasSleeping = this.monitor == null;
+
+		this.cached = true;
+		final TileEntity self = this.getHost().getTile();
+		final TileEntity target = self.getWorld().getTileEntity( self.getPos().offset( this.getSide().getFacing() ) );
+		final int newHandlerHash = this.createHandlerHash( target );
+
+		if( newHandlerHash != 0 && newHandlerHash == this.handlerHash )
+		{
+			return this.handler;
+		}
+
+		this.handlerHash = newHandlerHash;
+		this.handler = null;
+		this.monitor = null;
+		if( target != null )
+		{
+			IMEInventory<IAEFluidStack> inv = this.getInventoryWrapper( target );
+			if( inv instanceof ITickingMonitor )
+			{
+				this.monitor = (ITickingMonitor) inv;
+				this.monitor.setActionSource( new MachineSource( this ) );
+			}
+
+			if( inv != null )
+			{
+				this.checkInterfaceVsStorageBus( target, this.getSide().getOpposite() );
+
+				this.handler = new MEInventoryHandler<IAEFluidStack>( inv, AEApi.instance().storage().getStorageChannel( IFluidStorageChannel.class ) );
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Replace the type specification in this constructor call with the diamond operator ("<>"). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS2293)